pkgs/top-level/config.nix

```diff
@@ -123,6 +123,12 @@ let
       feature = "build packages with CUDA support by default";
     };
 
+    rocmSupport = mkMassRebuild {
+      type = types.bool;
+      default = false;
+      feature = "build packages with ROCm support by default";
+    };
```

Comment:
What's the purpose/rationale for this? Is it used by anything?

Reply:
I also unfortunately can't enable this by default, because CUDA and ROCm builds can and typically are mutually exclusive.

Reply:
I actually think having `-rocm` and `-cuda` variants might be preferrable, since that means you can choose between them without needing to re-evaluate all of nixpkgs.

`all-packages.nix` is already ginormous, I doubt the few packages that'll end up with `-cuda`/`-rocm` variants are what will tip it over the edge :D

Reply:
Maybe, but as long as hydra builds the rocm variants I fail to see the issue.
Same logic could be made to ``config.cudaSupport``, AFAIK.

Reply:
As-is Hydra won't build the rocm variants.

Config options are also not discoverable, people won't find them through `nix search`.

Reply:
cc @markuskowa since he's the one who suggested ``rocmSupport`` to me.
Any input you have would be helpful here.
Would it be better to have rocmSupport on by default and be turned off when cudaSupport is true?
We can do this because rocm stuff is open source with the exception of hsa-amd-aqlprofile which only one package may or may not actually need.

Reply:
Basically, rocmSupport = !cudaSupport, but we specify rocmSupport in the config anyway because just because we don't want to build CUDA doesn't mean we want to build ROCm.

Reply:
See 01bcd69ed8cb13f38c5fc7cd23a7250b5c594e49 for implementation.

Reply:
This allows us 3 states.
- cudaSupport
- rocmSupport
- none

Reply:
removed for now